virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/conversions/from_long/ToDouble.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.virtdata.library.basics.shared.conversions.from_long;
+
+import io.nosqlbench.virtdata.api.annotations.Categories;
+import io.nosqlbench.virtdata.api.annotations.Category;
+import io.nosqlbench.virtdata.api.annotations.ThreadSafeMapper;
+import io.nosqlbench.virtdata.api.bindings.VirtDataConversions;
+
+import java.util.function.LongToDoubleFunction;
+
+
+/**
+ * Create a double by converting values. This function works in the following modes:
+ * <UL>
+ *     <LI>If a double is provided, then the long input is scaled to be between 0.0f and the value provided,
+ *     as a fraction of the highest possible long.</LI>
+ *     <LI>If any other type of number is provided, then this function returns the equivalent double value.</LI>
+ *     <LI>Otherwise, the input is assumed to be a function which takes a long input, and, if necessary,
+ *     adapts to return a double with an appropriate type conversion.</LI>
+ * </UL>
+ */
+@Categories(Category.conversion)
+@ThreadSafeMapper
+public class ToDouble implements LongToDoubleFunction {
+
+    private final LongToDoubleFunction func;
+
+    ToDouble(Object func) {
+        if (func instanceof Double aDouble) {
+            double scale = aDouble.doubleValue();
+            this.func = l -> (double) (l % scale);
```

Comment:
Reading the comments:
```
 If a double is provided, then the long input is scaled to be between 0.0f and the value provided, as a fraction of the highest possible long.
 ```
Mod will get us between 0 and `scale`. But can you explain the `scaled...as a fraction of the highest possible long`part?

Reply:
The docs should def be improve here.

What this means is, that since the input is always a long, that is the "typed unit inteval" we have to work with, i.e. <long variate> / <maximum long value> as a fractional magnitude. Based on that, if you want to produce a double value, it will be by mapping (0.0d/(double) Long.MAX_VALUE) a a unit interval to a double.

This was an attempt to provide multiple options for how this ToDouble is used, but I think it is overly complicated, given that the variations do actually have different behavior. We should separate these.